crates/dbsp/src/tutorial.rs

```diff
@@ -0,0 +1,744 @@
+//! people_vaccinated       45346779
+//! people_fully_vaccinated 42832402
+//! total_boosters  60729479
+//! daily_vaccinations_raw  98496
+//! daily_vaccinations      126030
+//! total_vaccinations_per_hundred  251.87
+//! people_vaccinated_per_hundred   80.19
+//! people_fully_vaccinated_per_hundred     75.74
+//! total_boosters_per_hundred      107.39
+//! daily_vaccinations_per_million  2229
+//! daily_people_vaccinated 1139
+//! daily_people_vaccinated_per_hundred     0.002
+//! ```
+//!
+//! This is good enough to read the file.  We can combine Rust's `csv` crate to
+//! read CSV files with `serde` for deserializing into a `struct` and `time` for
+//! parsing the date field.  We're only going to work with the `location`,
+//! `date`, and `daily_vaccinations` fields, so we can omit the rest of the
+//! fields from the `Record` struct.  The full program is below and in
+//! `tutorial1.rs`:
+//!
+//! ```rust
+//! use anyhow::Result;
+//! use csv;
+//! use serde::Deserialize;
+//! use time::Date;
+//!
+//! #[allow(dead_code)]
+//! #[derive(Debug, Deserialize)]
+//! struct Record {
+//!     location: String,
+//!     date: Date,
+//!     daily_vaccinations: Option<u64>,
+//! }
+//!
+//! fn main() -> Result<()> {
+//!     let path = format!(
+//!         "{}/examples/tutorial/vaccinations.csv",
+//!         env!("CARGO_MANIFEST_DIR")
+//!     );
+//!     let mut reader = csv::Reader::from_path(path)?;
+//!     for result in reader.deserialize() {
+//!         let record: Record = result?;
+//!         println!("{:?}", record);
+//!     }
+//!     Ok(())
+//! }
+//! ```
+//!
+//! If we run this, then it prints the records in `Debug` format.  Here are the
+//! first few:
+//!
+//! ```text
+//! Record { location: "England", date: 2021-01-10, daily_vaccinations: None }
+//! Record { location: "England", date: 2021-01-11, daily_vaccinations: Some(140441) }
+//! Record { location: "England", date: 2021-01-12, daily_vaccinations: Some(164043) }
+//! Record { location: "England", date: 2021-01-13, daily_vaccinations: Some(192088) }
+//! Record { location: "England", date: 2021-01-14, daily_vaccinations: Some(213978) }
+//! ...
+//! ```
+//!
+//! ### Reading CSV into a DBSP circuit
+//!
+//! [`RootCircuit::build`] is the basic way to create a circuit in DBSP.
+//! A no-op use of it looks like this, which is also in `tutorial2.rs`:
+//!
+//! ```rust
+//! use anyhow::Result;
+//! use dbsp::RootCircuit;
+//!
+//! fn build_circuit(_circuit: &mut RootCircuit) -> Result<()> {
+//!     Ok(())
+//! }
+//!
+//! fn main() -> Result<()> {
+//!     let (_circuit, ()) = RootCircuit::build(build_circuit)?;
+//!     Ok(())
+//! }
+//! ```
+//!
+//! To start working with it, we can feed our records into a circuit through a
+//! "Z-set" ([`ZSet`]) input stream.  A "Z-set" is a set in
+//! which each item is associated with an integer weight.  In the context of
+//! changes to a data set, a positive weight represents an insertion and a
```

Comment:
Thanks, I changed this paragraph to:

```rust
//! To start working with it, we can feed our records into a circuit through a
//! "Z-set" ([`ZSet`]) input stream.  A "Z-set" is a set in which each item is
//! associated with an integer weight.  In the context of changes to a data set,
//! positive weights represent insertions and negative weights represent
//! deletions.  The magnitude of the weight represents a count, so that a weight
//! of 1 represents an insertion of a single copy of a record, 2 represents two
//! copies, and so on, and similarly for negative weights and deletions.  Thus,
//! a Z-set represents changes to a multiset.
```